src/cascade/model/dismod_groups.py

```diff
@@ -70,19 +71,32 @@ def check_alignment(self, other):
         """Check whether and where two DismodGroups are misaligned."""
         one_not_other = list()
         for group_name, group in self.items():
-            a_keys = list(group.keys())
-            b_keys = list(other[group_name].keys())
-            single_effect = len(a_keys) == 1 or len(b_keys) == 1
-            if group_name == "random_effect" and single_effect:
-                all_effects = a_keys[0][1] is None or b_keys[0][1] is None
-                if single_effect and all_effects:
-                    a_keys = [k[0] for k in a_keys]
-                    b_keys = [k[0] for k in b_keys]
-            a_keys = set(a_keys)
-            b_keys = set(b_keys)
-            if a_keys - b_keys:
-                one_not_other.append(f"left {group_name} has {a_keys - b_keys}")
-            elif b_keys - a_keys:
-                one_not_other.append(f"right {group_name} has {b_keys - a_keys}")
-            # else they agree
+            if group_name != "random_effect":
+                a_keys = list(group.keys())
```

Comment:
Bit weird to coerce to a list and then immediately coerce to a set

Reply:
removed the coercion and made it shorter.